File: src/main/java/com/elissamc/player/task/ElderGuardianTask.java
package com.elissamc.player.task;

import cn.nukkit.Player;
import cn.nukkit.level.Position;
import cn.nukkit.network.protocol.LevelEventPacket;
import cn.nukkit.scheduler.NukkitRunnable;
import com.elissamc.ElissaMC;

public class ElderGuardianTask extends NukkitRunnable {
	private final Player player;
	private final Position location;
	private final String title;
	private final String subtitle;

	public ElderGuardianTask(ElissaMC instance, Player player, Position location, String title, String subtitle) {
		this.player = player;
		this.location = location;
		this.title = title;
		this.subtitle = subtitle;
	}

	@Override
	public void run() {
		final LevelEventPacket pk = new LevelEventPacket();
		pk.evid = LevelEventPacket.EVENT_GUARDIAN_CURSE;
		pk.data = 0;
		pk.x = (float) player.x;
		pk.y = (float) player.y;
		pk.z = (float) player.z;

		player.dataPacket(pk);
		player.setSubtitle(subtitle);
		player.sendTitle(title);
		player.teleport(location);

	}

}

File: src/main/kotlin/com/elissamc/task/ElderGuardianTask.kt
package com.elissamc.task

import cn.nukkit.Player
import cn.nukkit.level.Position
import cn.nukkit.network.protocol.LevelEventPacket
import cn.nukkit.scheduler.NukkitRunnable

class ElderGuardianTask(private val player: Player, private val location: Position, private val title: String, private val subtitle: String) : NukkitRunnable() {

    override fun run() {
        val pk = LevelEventPacket()
        pk.evid = LevelEventPacket.EVENT_GUARDIAN_CURSE
        pk.data = 0
        pk.x = player.x.toFloat()
        pk.y = player.y.toFloat()
        pk.z = player.z.toFloat()

        player.dataPacket(pk)
        player.setSubtitle(subtitle)
        player.sendTitle(title)
        player.teleport(location)

    }

}
